packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -61,4 +56,96 @@ export function watchTransactionForConfirmations<
 			}
 		})() as unknown;
 	}, web3Context.transactionReceiptPollingInterval ?? web3Context.transactionPollingInterval);
+};
+
+const waitBySubscription = ({
+	web3Context,
+	transactionReceipt,
+	transactionPromiEvent,
+	returnFormat,
+}: WaitProps) => {
+	setImmediate(() => {
+		web3Context.subscriptionManager
+			?.subscribe('newHeads')
+			.then((subscription: NewHeadsSubscription) => {
+				subscription.on('data', async (data: BlockOutput) => {
+					if (!data?.number) {
+						return;
+					}
+					const confirmationNumber =
+						BigInt(data.number) - BigInt(transactionReceipt.blockNumber);
+					if (confirmationNumber >= web3Context.transactionConfirmationBlocks) {
+						transactionPromiEvent.emit('confirmation', {
+							confirmationNumber: format(
+								{ eth: 'uint' },
+								confirmationNumber,
+								returnFormat,
+							),
+							receipt: transactionReceipt,
+							latestBlockHash: format(
+								{ eth: 'bytes32' },
+								data.parentHash as HexString32Bytes,
+								returnFormat,
+							),
+						});
```

Comment:
This emission should be happening for every new block that comes after the one the transaction was included in (i.e. every confirmation). The only thing dependent on the check:

```typescript
if (confirmationNumber >= web3Context.transactionConfirmationBlocks)
```

is `await subscription.unsubscribe();`

Reply:
oh yes. thanks